plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererService.kt

```diff
@@ -753,6 +622,33 @@ class CodeWhispererService(private val cs: CoroutineScope) : Disposable {
         return states
     }
 
+    fun createInlineCompletionParams(
+        editor: Editor,
+        triggerTypeInfo: TriggerTypeInfo,
+        nextToken: Either<String, Int>?,
+    ): InlineCompletionWithReferencesParams =
+        ReadAction.compute<InlineCompletionWithReferencesParams, RuntimeException> {
+            InlineCompletionWithReferencesParams(
+                context = InlineCompletionContext(
+                    // Map the triggerTypeInfo to appropriate InlineCompletionTriggerKind
+                    triggerKind = when (triggerTypeInfo.triggerType) {
+                        CodewhispererTriggerType.OnDemand -> InlineCompletionTriggerKind.Invoke
+                        CodewhispererTriggerType.AutoTrigger -> InlineCompletionTriggerKind.Automatic
+                        else -> InlineCompletionTriggerKind.Invoke
+                    }
+                ),
+            ).apply {
+                textDocument = TextDocumentIdentifier(toUriString(editor.virtualFile))
+                position = Position(
+                    editor.caretModel.primaryCaret.logicalPosition.line,
+                    editor.caretModel.primaryCaret.logicalPosition.column
+                )
```

Comment:
dont we have a util for this already?

Reply:
I don't find one in the codebase
we have CaretPosition dataclass but that's for visualPosition, for poup displaying purpose.